common/classify.py

```diff
@@ -36,6 +38,11 @@ def generate_training(self):
                     continue
                 X = np.concatenate((X, part_X), axis=0)
                 Y = np.concatenate((Y, part_Y), axis=0)
+        if not found_training_file:
+            raise RuntimeError("Cannot find 'training.mat' files.")
+        if not X:
```

Comment:
Will the code ever hit the check on line 43? It seems to me that if found_training_file is True, then X will always have some elements in it. I may be missing something though.